app/src/main/java/com/xinto/opencord/ui/widget/Message.kt

```diff
@@ -27,64 +26,136 @@ import com.xinto.opencord.ui.component.OCAsyncImage
+                        WidgetBranchBTR(
+                            modifier = Modifier
+                                .height(16.dp)
+                                .width(36.dp),
+                        )
+                    }
+                    reply()
+                }
             }
-            Column(
-                modifier = Modifier.weight(1f),
-                verticalArrangement = Arrangement.spacedBy(4.dp),
+            Row(
+                horizontalArrangement = Arrangement.spacedBy(8.dp),
+                verticalAlignment = Alignment.Top
             ) {
+                if (avatar != null) {
+                    avatar()
+                }
                 Column(
-                    modifier = Modifier.fillMaxWidth(),
+                    modifier = Modifier.weight(1f),
+                    verticalArrangement = Arrangement.spacedBy(4.dp),
                 ) {
-                    if (author != null) {
-                        author()
-                    }
-                    if (content != null) {
-                        content()
-                    }
-                }
-                if (attachments != null) {
                     Column(
-                        modifier = Modifier.fillMaxWidth(0.9f),
-                        verticalArrangement = Arrangement.spacedBy(4.dp)
+                        modifier = Modifier.fillMaxWidth(),
                     ) {
-                        attachments()
+                        if (author != null) {
+                            author()
+                        }
+                        if (content != null) {
+                            content()
+                        }
                     }
-                }
-                if (embeds != null) {
-                    Column(
-                        modifier = Modifier.fillMaxWidth(0.9f),
-                        verticalArrangement = Arrangement.spacedBy(2.dp)
-                    ) {
-                        embeds()
+                    if (attachments != null) {
+                        Column(
+                            modifier = Modifier.fillMaxWidth(0.9f),
+                            verticalArrangement = Arrangement.spacedBy(4.dp)
+                        ) {
+                            attachments()
+                        }
+                    }
+                    if (embeds != null) {
+                        Column(
+                            modifier = Modifier.fillMaxWidth(0.9f),
+                            verticalArrangement = Arrangement.spacedBy(2.dp)
+                        ) {
+                            embeds()
+                        }
                     }
                 }
             }
         }
     }
 }
 
+@Composable
+fun WidgetMessageReplyContent(
+    text: AnnotatedString,
+    modifier: Modifier = Modifier,
+) {
+    ProvideTextStyle(MaterialTheme.typography.bodySmall) {
+        Text(
+            modifier = modifier,
+            text = text,
```

Comment:
The text should be combined into 1 line, otherwise the ellipsis overflow cuts it off incorrectly

Reply:
Impossible because of AnnotatedString